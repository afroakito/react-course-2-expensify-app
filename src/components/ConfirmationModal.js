import React from 'react';
import Modal from 'react-modal';

Modal.setAppElement('body');

const ConfirmationModal = (props) => (
    <Modal
        isOpen={props.confModal}
        onRequestClose={props.handleClearModal}
        contentLabel='Confirmation'
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Do you actually want to delete this expense?</h3>
        <div className="modal__button-wrapper">
            <button className="button accept" onClick={props.handleAcceptModal}>Okay</button>
            <button className="button reject" onClick={props.handleClearModal}>No</button>
        </div>
    </Modal>
);

export default ConfirmationModal;