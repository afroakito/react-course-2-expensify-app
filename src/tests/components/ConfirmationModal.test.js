import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationModal from '../../components/ConfirmationModal';

test('should correctly render ConfirmationModal', () => {
    const wrapper = shallow(<ConfirmationModal />);
    expect(wrapper).toMatchSnapshot();
});

