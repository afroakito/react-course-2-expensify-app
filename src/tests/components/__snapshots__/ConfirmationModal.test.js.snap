// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ConfirmationModal 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirmation"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    Do you actually want to delete this expense?
  </h3>
  <div
    className="modal__button-wrapper"
  >
    <button
      className="button accept"
    >
      Okay
    </button>
    <button
      className="button reject"
    >
      No
    </button>
  </div>
</Modal>
`;
